pygmt/datasets/earth_relief.py

```diff
@@ -169,38 +141,21 @@ def load_earth_relief(
     # Choose earth relief data prefix
     if use_srtm and resolution in land_only_srtm_resolutions:
         if data_source == "igpp":
-            earth_relief_prefix = "srtm_relief_"
+            dataset_prefix = "srtm_relief_"
         else:
             raise GMTInvalidInput(
                 f"The {data_source} option is not available if 'use_srtm=True'."
                 " Set data_source to 'igpp'."
             )
     else:
-        earth_relief_prefix = earth_relief_sources.get(data_source)
-
-    # different ways to load tiled and non-tiled earth relief data
-    # Known issue: tiled grids don't support slice operation
-    # See https://github.com/GenericMappingTools/pygmt/issues/524
-    if region is None:
-        if resolution not in non_tiled_resolutions:
-            raise GMTInvalidInput(
-                f"'region' is required for Earth relief resolution '{resolution}'."
-            )
-        fname = which(f"@{earth_relief_prefix}{resolution}{reg}", download="a")
-        grid = load_dataarray(fname, engine="netcdf4")
-    else:
-        grid = grdcut(f"@{earth_relief_prefix}{resolution}{reg}", region=region)
-
-    # Add some metadata to the grid
-    grid.name = "elevation"
-    grid.attrs["long_name"] = "elevation relative to the geoid"
-    grid.attrs["units"] = "meters"
-    grid.attrs["vertical_datum"] = "EMG96"
-    grid.attrs["horizontal_datum"] = "WGS84"
-    # Remove the actual range because it gets outdated when indexing the grid,
-    # which causes problems when exporting it to netCDF for usage on the
-    # command-line.
-    grid.attrs.pop("actual_range")
-    for coord in grid.coords:
-        grid[coord].attrs.pop("actual_range")
+        dataset_prefix = earth_relief_sources.get(data_source)
+
+    dataset_name = "earth_relief"
+    grid = _load_earth_dataset(
+        dataset_prefix=dataset_prefix,
+        dataset_name=dataset_name,
```

Comment:
Move `dataset_name` before `dataset_prefix` following the function definition:
```suggestion
        dataset_name=dataset_name,
        dataset_prefix=dataset_prefix,
```